nomos-services/mixnet/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+use mixnet::{config::Config, Message, Mixnet};
+use overwatch_rs::services::{
+    handle::ServiceStateHandle,
+    relay::RelayMessage,
+    state::{NoOperator, NoState},
+    ServiceCore, ServiceData, ServiceId,
+};
+use serde::{Deserialize, Serialize};
+use tokio::sync::{broadcast, mpsc, oneshot};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct MixnetServiceSettings {
+    config: Config,
+}
+
+pub struct MixnetService {
+    service_state: ServiceStateHandle<Self>,
+    mixnet: Mixnet,
+    outbound_msg_tx: mpsc::Sender<Message>,
+    inbound_msg_tx: broadcast::Sender<Message>,
+}
+
+#[derive(Debug)]
+pub enum MixnetMsg {
+    Send(mixnet::Message),
+    Subscribe(oneshot::Sender<broadcast::Receiver<Message>>),
+}
+
+impl RelayMessage for MixnetMsg {}
+
+impl ServiceData for MixnetService {
+    const SERVICE_ID: ServiceId = "Mixnet";
+    type Settings = MixnetServiceSettings;
+    type State = NoState<Self::Settings>;
+    type StateOperator = NoOperator<Self::State>;
+    type Message = MixnetMsg;
+}
+
+const CHANNEL_SIZE: usize = 100;
+
+#[async_trait::async_trait]
+impl ServiceCore for MixnetService {
```

Comment:
So, the idea is to use a mixnet service within the node? 
I have some questions about this:
* Is it not possible of having it as part of the network backend directly?
* What are the advantages of having it as a service?
    * In case is to plug/unplug the mixnet, or make it optional. Is it even possible?

Reply:
I'm sure we can have mixnet as a part of the network backend. Even in that case, it's possible to plug in/out the mixnet. 
At the beginning, I made it as a service since I didn't have a clear view how to integrate it with gossipsub. But now, I think that's more reasonable to put it into network backends.

Reply:
I’m also trying an approach that implements a custom libp2p transport as we discussed before.

Reply:
Removed `MixnetService`: [aac096e](https://github.com/logos-co/nomos-node/pull/288/commits/aac096ecbb4809409beb174da682293ad29c7069). But still we need to find the optimal structure after we agree on the approach first.